sw/device/rom_exts/manifest.md

```diff
@@ -0,0 +1,67 @@
+# ROM_EXT Manifest Format
+
+Based on the format outlined in [boot.md](./boot)
+
+```
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ───────────
+|                           Identifier                          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                                                               |
++                                                               +
+|                                                               |
++                           Signature                           +
+|                                                               |  Unsigned
++                          (Size TBD)                           +    Area
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ───────────
+|             Length            |      Signature Algorithm      |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ Signed Area
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                           Public Key                          +
+|                                                               |
++                          (Size TBD)                           +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                     ROM_EXT Image Version                     |
```

Comment:
Also consider adding a uint32 timestamp.

Reply:
Does this include the key id as well? Including it would eliminate the need to perform this computation in keys and signature module.

Reply:
I think the mask rom *has* to recalculate the key id, so that you cannot provide a key with a non-matching id, bypassing a check that the key itself is allowed to be used to authenticate a ROM_EXT image. Maybe my mental model of how key ids work is wrong though. cc @moidx?

Reply:
@modix left the following comment in the keys and signature doc:
> The key stored in the manifest is useful for verifying the integrity of the image in other contexts. For secure boot, please use the key stored in the ROM.

Based on this, the keys and signurate module just needs the key id.

Reply:
My understanding is the whole key is not stored in the ROM, just the key id. This is something I should discuss with @gkelly

Reply:
The public key is useful to have in the manifest so that the image can be verified by external tools. As a performance optimization, the implementation may provide a key id which the ROM can use to index the public key from ROM. This requires the signing tool to be in sync with the ROM.

As for the keys. They should be stored in the mask rom constants section, allocate space for the modulus and exponent. The exponent can be stored in uint32_t. We should allocate space for at least 5 keys. 

If we run into space constraints, then we can look at storing just the hashes of the public keys, but this will add an additional comparison to secure boot.